app/controllers/api_controller.rb

```diff
@@ -49,4 +50,47 @@ def page_preview
     }
   end
 
+  # Send content for a page
+  def pages_content
+    page = Page.find_by(id: params[:id])
+    unless page
+      return render json: { errors: [ { detail: "Unable to find page" } ] }, status: 404
+    end
+
+    # Get content
+    content = params[:content]
+    unless content
+      return render json: { errors: [ { detail: "content paramter not provided" } ] }, status: 400
+    end
+
+    # Check page content
+    page.custom_html(content)
+    changes = PollPage.perform(page: page)
```

Comment:
This setup is pretty confusing to me. `PollPage` sounds like a task that queries a url to see if there are changes. The fact that that behavior changes completely based on the state of the page (no url), is unexpected to me. I think we should split the PollPage task up in two components: a routine to fetch the latest content (currently in `Post#html`) and a routine to compare the latest content with what's saved (currently `PollPage#perform`). That way you could get away from the somewhat suspect state manipulation of `Post` instances, and just call the second routine (`CompareSnapshots` or something). 

The more I'm looking at this the more daunting this refactoring seems. The whole thing is written to be dependent on the state of the Page object, so trying to avoid manipulating that will require wholesale changes. Not ideal. I'm OK with not going down this road for now. I think it's a bigger project than should be attempted in this PR.